src/openzeppelin/token/erc1155/library.cairo

```diff
@@ -0,0 +1,601 @@
+            amounts=amounts,
+            data_len=data_len,
+            data=data,
+        );
+        return ();
+    }
+
+    func _burn{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        from_: felt, id: Uint256, amount: Uint256
+    ) {
+        alloc_locals;
+        with_attr error_message("ERC1155: burn from the zero address") {
+            assert_not_zero(from_);
+        }
+
+        // Check uints validity
+        _check_id(id);
+        with_attr error_message("ERC1155: amount is not a valid Uint256") {
+            uint256_check(amount);
+        }
+
+        // Deduct from burner
+        let (from_balance: Uint256) = ERC1155_balances.read(id, from_);
+        with_attr error_message("ERC1155: burn amount exceeds balance") {
+            let (new_balance: Uint256) = SafeUint256.sub_le(from_balance, amount);
+        }
+
+        ERC1155_balances.write(id, from_, new_balance);
+
+        let (operator) = get_caller_address();
+        TransferSingle.emit(operator=operator, from_=from_, to=0, id=id, value=amount);
+        return ();
+    }
+
+    func _burn_batch{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        from_: felt, ids_len: felt, ids: Uint256*, amounts_len: felt, amounts: Uint256*
+    ) {
+        alloc_locals;
+        with_attr error_message("ERC1155: burn from the zero address") {
+            assert_not_zero(from_);
+        }
+        with_attr error_message("ERC1155: ids and amounts length mismatch") {
+            assert ids_len = amounts_len;
+        }
+
+        // Recursive call
+        _burn_batch_iter(from_, ids_len, ids, amounts);
+        let (operator) = get_caller_address();
+        TransferBatch.emit(
+            operator=operator,
+            from_=from_,
+            to=0,
+            ids_len=ids_len,
+            ids=ids,
+            values_len=amounts_len,
+            values=amounts,
+        );
+        return ();
+    }
+
+    func _set_approval_for_all{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
+        owner: felt, operator: felt, approved: felt
+    ) {
+        // check approved is bool
+        with_attr error_message("ERC1155: approval is not boolean") {
+            assert approved * (approved - 1) = 0;
+        }
+
+        // caller/owner already checked non-0
+        with_attr error_message("ERC1155: setting approval status for zero address") {
+            assert_not_zero(operator);
+        }
+
+        with_attr error_message("ERC1155: setting approval status for self") {
+            assert_not_equal(owner, operator);
+        }
+
+        ERC1155_operator_approvals.write(owner, operator, approved);
+        ApprovalForAll.emit(owner, operator, approved);
+        return ();
+    }
+
+    func _set_uri{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(uri: felt) {
+        ERC1155_uri.write(uri);
+        // should it emit or follow solidity's inability?
```

Comment:
@frangio can you give some context on how much this feature is requested or not by the community?

Reply:
this is independent from users' requests. this comment was for myself before [resolving it cannot be emitted](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC1155/ERC1155.sol#L232-L250).